### PR TITLE
Revert "Bump github.com/prometheus/prometheus to 0.40.5 (#16734)"

### PR DESCRIPTION
**Pull request by codeboten**
Part of #16821 

This reverts commit afe46dc8c973cb3c8b9e1ea2da3f0dce14909ec1.

### CODE REVIEW

File: pkg/translator/loki/go.mod [hunk 1]
```diff
@@ -4,7 +4,7 @@ go 1.18
 
 require (
 	github.com/go-logfmt/logfmt v0.5.1
-	github.com/grafana/loki v1.6.2-0.20221206154352-fee23745328c
```
Comment (bogdandrutu) [inline pkg/translator/loki/go.mod:7]: If you cannot remove `replace github.com/go-kit/log => github.com/dannykopping/go-kit-log v0.2.2-0.20221002180827-5591c1641b6b` this is useless change.

**Comment by dashpole** [comment]
What issues did the version upgrade cause?

File: pkg/translator/loki/go.mod (hunk 1 above)
Comment (codeboten) [reply pkg/translator/loki/go.mod:7]: this replace will be removed once i take out promtail in the following PR

Comment (bogdandrutu) [reply pkg/translator/loki/go.mod:7]: Just try that in the "pkg/translator/loki", that does not depend on promtail. If you cannot do that there you will not be able to do anywhere.

**Comment by codeboten** [comment]
closing in favour of https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16822

**Pull request closed without merging**